integration_test/headless_test.go

```diff
@@ -16,6 +16,10 @@ import (
 )
 
 func (s *connectedSuite) Test_SuccessfullyInterceptsHeadlessService() {
+	// GKE Autopilot does not support NET_ADMIN containers which means headless services can't be intercepted
+	if itest.GetProfile(s.Context()) == itest.GkeAutopilotProfile {
+		s.T().SkipNow()
```

Comment:
Perhaps use `Skip` instead of `SkipNow` and pass the above comment as an argument?